app/controllers/api/v1/applied_coupons_controller.rb

```diff
@@ -35,6 +56,12 @@ def create_params
           :percentage_rate,
         )
       end
+
+      def valid_status?(status)
+        return false unless status
```

Comment:
Not sure we need this line, as `AppliedCoupon.statuses.key?(nil)` return `false` ;)

Reply:
nice catch, will remove it now!